charts/openstack-cluster/values.yaml

```diff
@@ -119,6 +119,39 @@ apiServer:
   # The port to use for the API server
   port: 6443
 
+
+# Set ignition based OS
+# ignitionBasedOS:
+# Ignition Based OS specific configuration. 
+ignitionKubeadmConfigSpec:
+  initConfiguration:
+    nodeRegistration:
+      name: ${COREOS_OPENSTACK_HOSTNAME}
+  joinConfiguration:
+    nodeRegistration:
+      name: ${COREOS_OPENSTACK_HOSTNAME}
+  preKubeadmCommands:
+    - export COREOS_OPENSTACK_HOSTNAME=${COREOS_OPENSTACK_HOSTNAME%.*}
+    - envsubst < /etc/kubeadm.yml > /etc/kubeadm.yml.tmp
+    - mv /etc/kubeadm.yml.tmp /etc/kubeadm.yml
+  format: ignition
+  ignition:
+    containerLinuxConfig:
+      additionalConfig: |
+        systemd:
+          units:
+          - name: coreos-metadata-sshkeys@.service
+            enabled: true
+          - name: kubeadm.service
+            enabled: true
+            dropins:
+            - name: 10-flatcar.conf
+              contents: |
+                [Unit]
+                Requires=containerd.service coreos-metadata.service
+                After=containerd.service coreos-metadata.service
+                [Service]
+                EnvironmentFile=/run/metadata/flatcar
```

Comment:
I was actually thinking more like all this stuff being in the `openstack-cluster.ignitionKubeadmConfigSpec`, so that we don't even have a separate variable here. This would be more consistent with the existing setup. Maybe that template should be called `openstack-cluster.flatcarKubeadmConfigSpec` in case there are other Ignition-based OSs that have different requirements to Flatcar, and instead of `.Values.ignitionBasedOS` we can have `.Values.osName` that defaults to `ubuntu`.

It might also involve moving anything that is Ubuntu-specific from `values.yaml` to a corresponding `openstack-cluster.ubuntuKubeadmConfigSpec` template.

The user can still customise the kubeadmConfigSpec for either Flatcar or Ubuntu via the existing hooks.

Reply:
Should we call it `osDistro` to align with the glance image convention?